### ISSUE TITLE
Add support for skipping too long lines in Framing.delimiter

### ISSUE THREAD

**Issue by jacobtardieu** [akka/akka#22385]
In Akka Stream, the `Framing` object allows to split a stream (e.g. a `FileIO`) in chunks.
More specifically, the `Framing.delimiter` method splits the stream when encountering a given _delimiter_.
```scala
def delimiter(delimiter: ByteString, maximumFrameLength: Int, allowTruncation: Boolean = false): Flow[ByteString, ByteString, NotUsed]
```

I've been using it to parse the lines of a file using the `\n` delimiter.
When a line is longer than the _maximumFrameLength_ which is passed as a parameter, the Flow fails the Stream.

It means than as soon as we encounter a line too long, the stream is failed and we have no way to read the remaining of the file.

To circumvent this issue, we could add another parameter to the `delimiter` method, e.g. `skipTooLongFrames` which, when set to true, would just make the Flow ignore too long frames and continue to read the stream.
The method would look like:
```scala
def delimiter(delimiter: ByteString, maximumFrameLength: Int, allowTruncation: Boolean = false, skipTooLongFrames: Boolean = false): Flow[ByteString, ByteString, NotUsed]
```

Is there another way to deal with this? If not, could we consider adding the parameter? Or is this going against some principles of Akka Stream?
(Note that the same question applies for the `lengthField` method of the same object.)

**Comment by patriknw** [comment]
@jacobtardieu I'm not sure I understand. Are you suggesting that when it reaches `maximumFrameLength` it would just drop that chunk and start a new chunk from that position? Isn't it a high risk that the frame data is corrupt if it's missing the delimiter?

**Comment by jacobtardieu** [comment]
@patriknw Not exactly :  when it reaches the `maximumFrameLength`, the internal buffer is emptied (because the chunk is too long, we're going to ignore it) but we continue until the next delimiter to start a new chunk at a correct position.

**Comment by patriknw** [comment]
ok, that makes more sense. Perhaps we should have a settings parameter so that this api can be evolved with more parameters without having to add more method overloads?

**Comment by jacobtardieu** [comment]
Yes good idea, I agree that a settings parameter is better than overloading. Would you like me to send a PR?

**Comment by patriknw** [comment]
yes, please

**Comment by jacobtardieu** [comment]
Ok thanks, I'll do it in the next few weeks.

**Comment by jacobtardieu** [comment]
After having coded the skipping part, I now have some issues:
* We seem to have no way to expose the fact that we're skipping a frame because it's too long: it means that we can't know we're skipping it (a way to log or to increment a metric would be nice).
* If we want to add a settings parameter (e.g. something called a `FramingSettings`) and make it work like the other settings parameters in Akka (I looked at `ActorMaterializerSettings`), we must be able to read the values from a config file, and for that we must have an ActorSystem in scope, which makes us lose the fact that `Framing` is a standalone object (which is precious especially in the javadsl).

Do you have any suggestion or advice on how I could circumvent these issues?

For now, the easiest way in my software was to copy-paste the `Framing.delimiter` part to add the skipping and the monitoring part (e.g. number of too long frames).

**Comment by hepin1989** [comment]
When the chunk is too long, there are two cases:
- drop the frame and return an error/response
- fail the stream
